src/openapi/streaming/connection/transport/websocket-transport.ts

```diff
@@ -526,57 +544,51 @@ class WebsocketTransport implements StreamingTransportInterface {
         };
         const url = `${this.authorizeUrl}?contextId=${contextId}`;
 
-        this.authorizePromise = fetch('put', url, options)
-            .then((response) => {
+        fetch('put', url, options).then(
+            () => {
                 log.debug(LOG_AREA, 'Authorization completed', {
                     contextId,
                 });
-                return response;
-            })
-            .catch((error) => {
+                // if this call was superseded by another one, then ignore this
+                if (
+                    this.authToken !== authToken ||
+                    this.contextId !== contextId
+                ) {
+                    return;
+                }
+                this.isAuthorized = true;
+                this.onAuthorized();
+            },
+            (error) => {
                 // if this call was superseded by another one, then ignore this error
                 if (
                     this.authToken !== authToken ||
                     this.contextId !== contextId
                 ) {
-                    return Promise.reject(
-                        new Error(
-                            'Failed websocket-transport authorize with a different token',
-                        ),
-                    );
+                    return;

```

Comment:
This was strange that this check is in the catch but not the then... it doesn't need a error because it isn't a error scenario